cupyx/scipy/sparse/_index.py

```diff
@@ -0,0 +1,388 @@
+"""Indexing mixin for sparse matrix classes.
+"""
+
+import cupy
+from cupy import core
+
+import numpy
+
+
+_int_scalar_types = (int, numpy.integer)
+
+
+def _get_csr_submatrix(Ap, Aj, Ax,
+                       start_maj, stop_maj,
+                       start_min, stop_min):
+    """Return a submatrix of the input sparse matrix by
+    slicing both major and minor axes.
+
+    Args
+        Ap : indptr array from input sparse matrix
+        Aj : indices array from input sparse matrix
+        Ax : data array from input sparse matrix
+        start_maj : starting index of major axis
+        stop_maj : ending index of major axis
+        start_min : starting index of minor axis
+        stop_min : ending index of minor axis
+
+    Returns
+        Bp : indptr array of output sparse matrix
+        Bj : indices array of output sparse matrix
+        Bx : data array of output sparse matrix
+    """
+
+    Bp = Ap[start_maj:stop_maj+1]
+
+    start_offset = Bp[0]
+    stop_offset = Bp[-1]
+
+    Aj = Aj[start_offset:stop_offset]
+    Ax = Ax[start_offset:stop_offset]
+
+    Aj_copy = cupy.empty(Aj.size+1, dtype=Aj.dtype)
```

Comment:
I know I suggested using `empty` here...Do you think it is safe to not set `Aj_copy[0] = 0`?

Reply:
`Aj_copy -= Aj_copy[0]` in L55 seems to recover the lack of initialization, but I think it is better to set `Aj_copy[0] = 0` here.